### PR TITLE
Added gentoo to installer

### PR DESCRIPTION
**Pull request by zjpxshade**
I think that the changes I added to /util/linux_install.sh should be enough to get this working on a Gentoo system. I have just successfully used it to update the keymap on my Iris.

### CODE REVIEW

**Comment by drashna** [comment]
@zjpxshade you may want to create a new branch here, and remove the last commit, so that this PR only includes the relevant changes.

File: util/linux_install.sh [hunk 1]
```diff
@@ -18,9 +17,24 @@ elif grep ID /etc/os-release | grep -q arch; then
...
+  echo "This script will make a USE change in order to ensure that that QMK works on your system. All changes will be sent to the the file /etc/portage/package.use/qmk_firmware -- please review it, and read Portage's output carefully before installing any packages on your system. You will also need to ensure that your kernel is compiled with support for the keyboard chip that you are using (e.g. enable Arduino for the Pro Micro). Further information can be found on the Gentoo wiki."
+  echo -n "Proceed (y/N)? "
+  old_stty_cfg=$(stty -g)
+  stty raw -echo
+  answer=$( while ! head -c 1 | grep -i '[ny]' ;do true ;done )
+  stty $old_stty_cfg
+    if echo "$answer" | grep -iq "^y" ;then
+      sudo touch /etc/portage/package.use/qmkfirmware
+      echo "sys-devel/gcc multilib" | sudo tee --append /etc/portage/package.use/qmkfirmware > /dev/null
+      sudo emerge -auN sys-devel/gcc app-arch/unzip app-arch/zip net-misc/wget app-mobilephone/dfu-util sys-devel/crossdev dev-embedded/avrdude
```
Comment (johu) [inline util/linux_install.sh]: If I am not mistaken dev-embedded/dfu-programmer is needed to flash the firmware on many boards.

**Comment by zjpxshade** [comment]
@drashna I have reverted the other commit and made a new branch -- thanks! I learn something new about git all the time. 

@johu That may be true, I can add it. I only have a ProMicro board, so I was wary of pulling packages I didn't need and having them be unnecessary or worse break the thing. Since many people use Gentoo to get even less bloat than normal, it's prabably fair to ask Gentoo users to uncomment the packages they need themselves; idk.

**Comment by drashna** [comment]
Git can/is be super confusing and very powerful.

In theory, you could rebase with `-i` and remove the pick line for that commit.  It would remove it from the branch.  But this is a bit more advanced, and I'd say play with it on a test branch first.

Also, it's good practice to leave the master branch alone, (other than to keep it up to date), and create branches from it for each new feature/PR.   That way, you don't have to worry about this sort of thing in the future.

**Comment by johu** [comment]
@zjpxshade You are in luck I am not only a Gentoo user but a developer as well and I used dfu-programmer for clueboard and dz60 successfully.

**Comment by zjpxshade** [comment]
@johu I'm not sure whether to put it in my branch or this one now but either way it's in one of the versions of the files. Thanks!

**Comment by jackhumbert** [comment]
Awesome - thanks!

**Pull request merged**